CMakeLists.txt

```diff
@@ -38,29 +36,33 @@ CPMAddPackage(
 )
 
 # ---- Add source files ----
-
-# Note: globbing sources is considered bad practice as CMake's generators may not detect new files
-# automatically. Keep that in mind when changing files, or explicitly mention them here.
-file(GLOB_RECURSE headers CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/include/*.h")
-file(GLOB_RECURSE sources CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/source/*.cpp")
+set(include_dirs
```

Comment:
Is it better practice to name files explicitly instead of using GLOB_RECURSE?

Reply:
Yes. Standard CMake practices recommend against globbing for a few reasons. Chiefly because it won't pick up adding/removing files- although that can be mitigated with a generator (it essentially pushes the globbing into the Makefile/Ninja/etc.). Globbing the headers also prevents using the subdirectory names in includes (i.e., #include "parser/foo" instead of #include "foo").

Reply:
Ah ok. Noted!